pandas/formats/format.py

```diff
@@ -2060,7 +2060,19 @@ def get_result_as_array(self):
         """
 
         if self.formatter is not None:
-            return np.array([self.formatter(x) for x in self.values])
+            if self.na_rep is None:
```

Comment:
you are repeating lots of code from right below ``format_values_with``. see if you can simplify